adding documentation for the new export button (#2578)
o3de/o3de.org: content/docs/user-guide/packaging/project-export/project-export-android.md

This guide covers how to use project export tooling to build and deploy games on Android. We will use the Android export script to generate a Gradle project, prepare assets, and then build the project. The end result of export should be an APK file that can be installed on your Android device, along with an additional step to auto-install on a connected phone.

{{< note >}}

> To learn more about the project export tooling in general, please consult the page [Project Export for Windows and Linux.](/docs/user-guide/packaging/project-export/project-export-pc)

Reviewer: Why `in general`? It was fine before

> To learn more about the project export tooling in general, please consult the page [Project Export for Windows and Linux.](/docs/user-guide/packaging/project-export/project-export-pc)

Reviewer 2: ```suggestion
To learn more about the project export tooling in general, please consult the page [Project Export for Windows and Linux](/docs/user-guide/packaging/project-export/project-export-pc).
```

{{< /note >}}
{{< note >}}

> To learn more about how to manually export an Android project, please consult the page [Generating an Android Project](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/generating_android_project_windows/#prerequisites).

Reviewer 2: Notice that there are multiple places still using `development--o3deorg.netlify.app` in links. Maybe we update all to relative paths?

For this one, it would be like this:
```suggestion
To learn more about how to manually export an Android project, please consult the page [Generating an Android Project](/docs/user-guide/packaging/project-export-android/#prerequisites).
```

But I am not sure if `#prerequisites` can work here.

{{< /note >}}

## Prerequisites
1. Make sure that the [Project Export for Windows and Linux page](/docs/user-guide/packaging/project-export/project-export-pc) prerequisites are satisfied.

> 2. Make sure you have your project created and registered with your engine. Check that you can successfully build and open your project with the Project Manager and Editor.

Reviewer: `Project Manager` should be removed now
```suggestion
2. Make sure you have your project created and registered with your engine. Check that you can successfully build and open your project with the Editor.
```

> 3. Ensure all pre-requisites for working with Android in O3DE are satsified. You can learn about software dependencies [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/#prerequisite-software-and-packages) and project setup pre-requisites [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/generating_android_project_windows/#prerequisites). For the Android SDK, make sure to record where the root folder path is located on your hard drive, as that will be needed for setting up configuration.

Reviewer 2: ```suggestion
3. Ensure all prerequisites for working with Android in O3DE are satsified. You can learn about software dependencies [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/#prerequisite-software-and-packages) and project setup pre-requisites [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/generating_android_project_windows/#prerequisites). For the Android SDK, make sure to record where the root folder path is located on your hard drive, as that will be needed for setting up configuration.
```

> 3. Ensure all pre-requisites for working with Android in O3DE are satsified. You can learn about software dependencies [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/#prerequisite-software-and-packages) and project setup pre-requisites [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/generating_android_project_windows/#prerequisites). For the Android SDK, make sure to record where the root folder path is located on your hard drive, as that will be needed for setting up configuration.

Reviewer 2: ```suggestion
3. Ensure all prerequisites for working with Android in O3DE are satsified. You can learn about software dependencies [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/#prerequisite-software-and-packages) and project setup pre-requisites [here](https://development--o3deorg.netlify.app/docs/user-guide/platforms/android/generating_android_project_windows/#prerequisites). For the Android SDK, make sure to record where the root folder path is located on your hard drive, as that will be needed for setting up configuration.
```

Same comment. Maybe use prerequisites in all places.

```
Make sure the passwords supplied here match what you used in the keystore file, and that you remember what they are.

## Using Project Manager
### Getting Started
To learn more about the export button feature and the export settings panel in the Project Manager in general, [click here](project-export-pc/#getting-started). This guide will only cover Android specific details.

> To kick off Android export, the process is identical to what would be done for Windows or Linux. From your project card, simply click the drop-down and go to `Export Launchers->Android`. When clicked, after confirming you are ready to proceed, the export should immediately begin.

Reviewer 2: For consistency, use either "dropdown" or "drop-down".

```suggestion
To kick off Android export, the process is identical to what would be done for Windows or Linux. From your project card, simply open the dropdown menu and click the Android option in the Export Launchers sub-menu. When clicked, after confirming you are ready to proceed, the export should immediately begin.
```

The Android specific features are as follows:

* Deploy to Android Device - upon completion of the export process, if you have an Android device connected to your computer (and `adb` can recognize the device), then the exported Android app will be automatically installed onto your device. This can be useful in quickly testing on real hardware.
* Asset Mode Configuration - specifies how game assets should be organized inside the APK file bundle. LOOSE means that all files are stored individually, acting like a primitive zip archive. PAK means that all files are combined into a single blob file.
* Android APK Build Path - specifies where the Android project should be generated, which will inform gradle on where to build the final APK.

{{< note >}}
On some platforms, such as Windows, when trying to export an Android APK project, you may run into an issue with Gradle prematurely failing  the export because of something like:
```
...the filename is too long (exceeds 260 characters)
```

> If you encounter this issue, and the Android APK build path is relative to the project folder, you may need to change the folder closer to the root of your hard drive (such as `C:/o3de-android`).

Reviewer 2: ```suggestion
If you encounter this issue, and the Android APK build path is relative to the project folder, you may need to move the folder closer to the root of your hard drive such as `C:/o3de-android`.
```

## Android Export Script
O3DE ships with an [Android Export Script](https://github.com/o3de/o3de/blob/bb3eafe30d8291f50b69924e5b7a432c8c6f53ca/scripts/o3de/ExportScripts/export_source_android.py#L28), capable of generating an Android Gradle project folder to handle standard use cases of O3DE projects on Android.

> The export script has two primary sections: the function [`export_source_android_project`](https://github.com/o3de/o3de/blob/bb3eafe30d8291f50b69924e5b7a432c8c6f53ca/scripts/o3de/ExportScripts/export_source_android.py#L28) and the [startup code that only runs if the script is invoked by the CLI](https://github.com/o3de/o3de/blob/bb3eafe30d8291f50b69924e5b7a432c8c6f53ca/scripts/o3de/ExportScripts/export_source_android.py#L222). In-depth discussion on these two sections can be found in the [Developer Guide](/docs/engine-dev/tools/o3de-cli/project-export).

Reviewer 2: ```suggestion
The export script has two primary sections: the function [`export_source_android_project`](https://github.com/o3de/o3de/blob/bb3eafe30d8291f50b69924e5b7a432c8c6f53ca/scripts/o3de/ExportScripts/export_source_android.py#L28) and the [startup code](https://github.com/o3de/o3de/blob/bb3eafe30d8291f50b69924e5b7a432c8c6f53ca/scripts/o3de/ExportScripts/export_source_android.py#L222) that only runs if the script is invoked by the CLI. In-depth discussion on these two sections can be found in the [Developer Guide](/docs/engine-dev/tools/o3de-cli/project-export).
```
